crates/ruff_linter/src/directives.rs

```diff
@@ -335,11 +335,21 @@ impl FromStr for TodoDirectiveKind {
 
     fn from_str(s: &str) -> Result<Self, Self::Err> {
         // The lengths of the respective variant strings: TODO, FIXME, HACK, XXX
+
         for length in [3, 4, 5] {
             let Some(substr) = s.get(..length) else {
                 break;
             };
 
+            if s.len() > length {
```

Comment:
Nit: I would change the `from_str` implementation to do a direct match on the string (instead of selecting the substring) and instead split by word boundaries here:

https://github.com/astral-sh/ruff/blob/8f409285347fbed17c0764759635eac6b6935b13/crates/ruff_linter/src/directives.rs#L302

That should simplify the implementation

Reply:
I don't quite understand, sorry (Maybe my Rust knowledge is not good enough)

Reply:
I'm proposing a small rewrite because I think it will simplify the overall code now that we make the change here. 

What I propose is that `FromStr` does a full match (matching the entire string and not just a prefix). So `from_str` becomes something like this:


```rust
            match s.to_lowercase().as_str() {
                "fixme" => {
                    Ok(TodoDirectiveKind::Fixme)
                }
                "hack" => {
                    Ok(TodoDirectiveKind::Hack)
                }
                "todo" => {
                    Ok(TodoDirectiveKind::Todo)
                }
                "xxx" => {
                    Ok(TodoDirectiveKind::Xxx)
                }
                _ =>Err(()),
		}
```

This does require that we change the one call site to implement the: split by word boundaries (whitespace) and we only pass the first word to `FromStr`

Reply:
I have applied your suggestion